sdk/nodejs/runtime/rpc.ts

```diff
@@ -309,13 +313,17 @@ export async function serializeProperty(ctx: string, prop: Input<any>, dependent
     }
 
     if (CustomResource.isInstance(prop)) {
-        // Resources aren't serializable; instead, we serialize them as references to the ID property.
+        // Resources aren't serializable; instead, we serialize them as references to the URN property.
         if (excessiveDebugOutput) {
-            log.debug(`Serialize property [${ctx}]: custom resource id`);
+            log.debug(`Serialize property [${ctx}]: custom resource urn`);
         }
 
         dependentResources.add(prop);
-        return serializeProperty(`${ctx}.id`, prop.id, dependentResources);
```

Comment:
this will straight up break things right?  (unless downstream you use the urn to get the id for serialization purposes)?

Reply:
Yes - this is a breaking change (and leads to some of the tests that are changed elsewhere).  We will need to figure out how to undo this (make it optional and only use it on the new serialization paths) if we want to land prior to a breaking change release (2.0).